### PR TITLE
Cassandra 16905

### PR DESCRIPTION
**Pull request by jmckenzie-dev**

### CODE REVIEW

File: src/java/org/apache/cassandra/db/marshal/CollectionType.java [hunk 1]
```diff
@@ -199,6 +199,15 @@ public boolean isValueCompatibleWithInternal(AbstractType<?> previous)
         return isValueCompatibleWithFrozen(tprev);
     }
 
+    @Override
+    public boolean isSerializationCompatibleWith(AbstractType<?> previous)
+    {
+        if (!isValueCompatibleWith(previous))
+            return false;
+
+        return valueComparator().isSerializationCompatibleWith(((CollectionType)previous).valueComparator());
```
Comment (maedhroz) [inline src/java/org/apache/cassandra/db/marshal/CollectionType.java:208]:
```suggestion
        return valueComparator().isSerializationCompatibleWith(((CollectionType<?>)previous).valueComparator());
```

File: test/distributed/org/apache/cassandra/distributed/test/NetstatsRepairStreamingTest.java [hunk 2]
```diff
@@ -76,7 +76,7 @@ private void executeTest(boolean compressionEnabled) throws Exception
 
             final Future<NetstatResults> resultsFuture1 = executorService.submit(new NetstatsCallable(node1));
 
-            node1.nodetoolResult("repair", "netstats_test").asserts().success();
+            node1.nodetoolResult("repair", "--full", "netstats_test").asserts().success();
```
Comment (maedhroz) [inline test/distributed/org/apache/cassandra/distributed/test/NetstatsRepairStreamingTest.java:79]: Is this related to the patch?

**Review by maedhroz — APPROVED** [review]

File: src/java/org/apache/cassandra/db/marshal/CollectionType.java (hunk 1 above)
Comment (maedhroz) [reply src/java/org/apache/cassandra/db/marshal/CollectionType.java:208]: There are actually a few other places in `CollectionType` we could get rid of compile warnings (and one in `ReversedType`).

File: test/distributed/org/apache/cassandra/distributed/test/NetstatsRepairStreamingTest.java (hunk 2 above)
Comment (jmckenzie-dev) [reply test/distributed/org/apache/cassandra/distributed/test/NetstatsRepairStreamingTest.java:79]: Nope - slipped through from another diff. I'll fix.

File: src/java/org/apache/cassandra/db/marshal/CollectionType.java (hunk 1 above)
Comment (jmckenzie-dev) [reply src/java/org/apache/cassandra/db/marshal/CollectionType.java:208]: :+1: Changed on branch. Thanks.
